### ISSUE TITLE
makefiles for BSD make utilities

### ISSUE THREAD

**Issue by requiredregistration** [pasky/pachi#158]
i checked this on openbsd and this is the result after entering `gmake`:

```
# gmake
/bin/sh: ./genbuild: No such file or directory
gmake: *** [Makefile:320: build.h.git] Error 1
```

**Comment by lemonsqueeze** [comment]
Hi,
Thanks for your input, this definitely shouldn't be happening.

Could you give more details about what you did ?
I'm assuming you're working from a git repository, as in :
```
$ git clone https://github.com/pasky/pachi
$ cd pachi
$ gmake
```

is that right ?

Output of this would help too:
```
$ ls -l genbuild
```

**Comment by lemonsqueeze** [comment]
Do you have bash installed in `/bin/bash` ?

**Comment by lemonsqueeze** [comment]
> what changed the output was the installation of bash. it uses bash for some things?

Yes, build needs a number of GNU tools. On OpenBSD the problem is that invoking `make` runs bsd make instead of gnu make, which won't work here. So you'll either have to hack your environment or Pachi's build to work around that. You could try replacing every instance of `@make` by `@$(MAKE)` in Makefile, and build with :
```
$ gmake MAKE=gmake
```

If you're serious about trying to get it to build on OpenBSD first you should have all dependencies listed in [INSTALL.md](https://github.com/pasky/pachi/blob/master/INSTALL.md) installed. This assumes you have a minimal setup for compiling software (`build-essential` package on Ubuntu). Here at the very least this means gmake, bash, gcc, g++, all dependencies + library headers, probably perl as well. It's pointless to even try building until you have all dependencies installed (boost, caffe etc ...).
